plugin_gems.rb

```diff
@@ -28,6 +28,8 @@
 download "fluent-plugin-multi-format-parser", "0.1.1"
 download "fluent-plugin-prometheus", "0.3.0"
 download "fluent-plugin-record-reformer", "0.9.1"
+download "fluent-plugin-record-modifier", "2.0.1"
```

Comment:
Why add `record_modifier` and not reuse `record-reformer`, which we already have? Or even use `record_transformer`, which comes with fluentd?

Reply:
We had to use some complicated logic (e.g. prefix labels with `k8s-pod-labels/`) that neither of the other two gems support.

Reply:
Example:
```
    <filter reform.**>
      @type record_modifier
      enable_ruby true
      <record>
        # Extract "kubernetes"->"labels" and set them as
        # "logging.googleapis.com/labels". Prefix these labels with
        # "k8s-pod-labels" to distinguish with other labels and avoid
        # label name collision with other types of labels.
        _dummy_ ${if record.is_a?(Hash) && record.has_key?('kubernetes') && record['kubernetes'].has_key?('labels') && record['kubernetes']['labels'].is_a?(Hash); then; record["logging.googleapis.com/labels"] = record['kubernetes']['labels'].map{ |k, v| ["k8s-pod-labels/#{k}", v]}.to_h ; end; nil}
      </record>
      # Delete rest of "kubernetes" and "docker".
      remove_keys _dummy_,kubernetes,docker
    </filter>
```

Reply:
The same should work as-is with both `record_transformer` and `record_reformer`, neither of which requires a new dependency... Did you try it?

Reply:
Yeah, I tried. `record` did not get updated when using those two plugins.

I think the problem is that they both pass `record` by value (https://github.com/fluent/fluentd/blob/c4d0a88b7f6b2772be66f84d5b6590cd03ef5310/lib/fluent/plugin/filter_record_transformer.rb#L131 and https://github.com/sonots/fluent-plugin-record-reformer/blob/0d8a36f8eb252209758c567cd2fe7a7b80d8b0e9/lib/fluent/plugin/out_record_reformer/core.rb#L136), while `record_modifier` pass `record` by reference (https://github.com/repeatedly/fluent-plugin-record-modifier/blob/1e6ecc946ed7f50ba489ddf5b61e05803a269cc1/lib/fluent/plugin/out_record_modifier.rb#L129).

Reply:
Ah. Got it. Sigh, we can clean it up later.